### ISSUE TITLE
Remap modifiers on all layers

### ISSUE THREAD

**Issue by mondalaci** [UltimateHackingKeyboard/agent#766]
According to the feedback we receive, users are not conscious about remapping modifiers on all layers which causes unwanted behavior on their end and extra support requests on our end.

In order to solve this, make Agent watch the state changes of the scancode and modifiers. If the scancode is None AND if any modifier is checked then automatically check the "Remap on all layers" checkbox.

If the checkbox gets automatically checked by Agent while the popover is opened, make Agent update the state of the checkbox afterwards, but only while the popover is open.

**Comment by mhantsch** [comment]
Not sure if this has anything to do with this issue, but I noticed the following (under Windows 10):

I have Agent open and in the foreground. I now use Alt-Tab to switch to a different Windows app. I then use Alt-Tab again to switch back to Agent. I have released the Alt key. I click on a key in Agent because I want to reconfigure it. The "remap on all layers" option now shows up checked. But I had not depressed the Alt key during the mouse click! I often do not notice this and I end up remapping keys on all layers which was really not my intention.

This is with Agent 1.2.8.
